_posts/2023-08-04-deploying-OIDC-webapp-on-openshift.adoc

```diff
@@ -0,0 +1,161 @@
+* https://quay.io/repository/[Quay]
+* https://helm.sh/[Helm]
+
+=== Logging into OpenShift CLI
+
+If you are not logged into OpenShift on your cli, navigate to Developer Sandbox, click on your username at the top right corner, copy your login command and paste it on your terminal. 
+
+Once you have your environment set up with the required tools, we can move on to the next step to build and deploy our application on OpenShift. 
+
+== Setting up Red Hat Single Sign On 
+Developer Sandbox gives us access to the enterprise version of Red Hat's OpenID provider, Red Hat Single Sign On (RH-SSO). On the topology view of OpenShift, click on the `Add to Project` button on the top left and add the version of RH-SSO that is compatible with the current version of OpehShift you are running and wait for it to finish deploying. Please refer to the https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.2/html/red_hat_single_sign-on_for_openshift/introduction[RH-SSO for OpenShift Documentation] to learn more about it.
+
+Next click on the `Open URL` button on the pod to be redirected to the RH-SSO landing page and click o `Administration Console`. You will be able to sign in using the admin username and password that is specified in the yaml configuration of the RH-SSO pod. 
+
+From here, create a realm called `myrealm`, and a client called `myclient`. 
+
+After registering our client, `myclient`, we also need to configure `valid redirect URIs`. Simply click on `Clients` and then on `myclient`. In the `Valid Redirect URIs` field, enter `*`.
+
+Navigate to the `Client Scopes` tab for `myclient`. Here you will see that some scope values are under `Assigned Default Client Scopes` field and some others are under `Assigned Optional Client Scopes` field. When client scopes are added to `Default`, the claims associated with them are automatically added to the access token you receive without the need to configure them. Changing it to `Optional` ensures that they are not included by default and you will only get access to claims if you configure the corresponding scope values. This list of scope values you see here are the values that RH-SSO allows and it varies for different OpenID providers. RH-SSO also allows you to create new scopes, but we will not be doing that for this blog. 
+
+Now, click on `Roles` and create two roles, `user` and `admin`.
+
+Finally, create a user called `alice` whose `First Name` is Alice and `Last Name` is Smith and assign her the `user` and `admin` roles. Next, navigate to the `Credentials` tab and create a password for Alice. Toggle the `Temporary` switch off so you are not prompted to update the password after the first login. 
+
+== Building Container Image for WildFly
+
+In order to deploy our application on OpenShift, we need to first create a container image of the application using Docker or Podman. In this post we are using Docker, but you can also use Podman using the `podman` keyword, or using an alias.  
```

Comment:
Here, it would be good to explain why we need to build WildFly first (since the feature hasn't been merged yet).

Reply:
Fixed!